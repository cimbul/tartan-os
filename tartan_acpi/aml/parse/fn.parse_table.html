<!DOCTYPE html><html lang="en"><head><meta charset="utf-8"><meta name="viewport" content="width=device-width, initial-scale=1.0"><meta name="generator" content="rustdoc"><meta name="description" content="API documentation for the Rust `parse_table` fn in crate `tartan_acpi`."><meta name="keywords" content="rust, rustlang, rust-lang, parse_table"><title>tartan_acpi::aml::parse::parse_table - Rust</title><link rel="stylesheet" type="text/css" href="../../../normalize.css"><link rel="stylesheet" type="text/css" href="../../../rustdoc.css" id="mainThemeStyle"><link rel="stylesheet" type="text/css" href="../../../light.css"  id="themeStyle"><link rel="stylesheet" type="text/css" href="../../../dark.css" disabled ><link rel="stylesheet" type="text/css" href="../../../ayu.css" disabled ><script id="default-settings"></script><script src="../../../storage.js"></script><noscript><link rel="stylesheet" href="../../../noscript.css"></noscript><link rel="icon" type="image/svg+xml" href="../../../favicon.svg">
<link rel="alternate icon" type="image/png" href="../../../favicon-16x16.png">
<link rel="alternate icon" type="image/png" href="../../../favicon-32x32.png"><style type="text/css">#crate-search{background-image:url("../../../down-arrow.svg");}</style></head><body class="rustdoc fn"><!--[if lte IE 8]><div class="warning">This old browser is unsupported and will most likely display funky things.</div><![endif]--><nav class="sidebar"><div class="sidebar-menu">&#9776;</div><a href='../../../tartan_acpi/index.html'><div class='logo-container rust-logo'><img src='../../../rust-logo.png' alt='logo'></div></a><div class="sidebar-elems"><p class="location"><a href="../../index.html">tartan_acpi</a>::<wbr><a href="../index.html">aml</a>::<wbr><a href="index.html">parse</a></p><div id="sidebar-vars" data-name="parse_table" data-ty="fn" data-relpath=""></div><script defer src="sidebar-items.js"></script></div></nav><div class="theme-picker"><button id="theme-picker" aria-label="Pick another theme!" aria-haspopup="menu"><img src="../../../brush.svg" width="18" alt="Pick another theme!"></button><div id="theme-choices" role="menu"></div></div><script src="../../../theme.js"></script><nav class="sub"><form class="search-form"><div class="search-container"><div><select id="crate-search"><option value="All crates">All crates</option></select><input class="search-input" name="search" disabled autocomplete="off" spellcheck="false" placeholder="Click or press ‘S’ to search, ‘?’ for more options…" type="search"></div><button type="button" class="help-button">?</button>
                <a id="settings-menu" href="../../../settings.html"><img src="../../../wheel.svg" width="18" alt="Change settings"></a></div></form></nav><section id="main" class="content"><h1 class="fqn"><span class="in-band">Function <a href="../../index.html">tartan_acpi</a>::<wbr><a href="../index.html">aml</a>::<wbr><a href="index.html">parse</a>::<wbr><a class="fn" href="">parse_table</a></span><span class="out-of-band"><span id="render-detail"><a id="toggle-all-docs" href="javascript:void(0)" title="collapse all docs">[<span class="inner">&#x2212;</span>]</a></span><a class="srclink" href="../../../src/tartan_acpi/aml/parse.rs.html#42-75" title="goto source code">[src]</a></span></h1><pre class="rust fn">pub fn parse_table(<br>&nbsp;&nbsp;&nbsp;&nbsp;data: <a class="primitive" href="https://doc.rust-lang.org/nightly/std/primitive.slice.html">&amp;[</a><a class="primitive" href="https://doc.rust-lang.org/nightly/std/primitive.u8.html">u8</a><a class="primitive" href="https://doc.rust-lang.org/nightly/std/primitive.slice.html">]</a><br>) -&gt; <a class="enum" href="https://doc.rust-lang.org/nightly/core/result/enum.Result.html" title="enum core::result::Result">Result</a>&lt;<a class="struct" href="../../../tartan_acpi/aml/struct.AMLTable.html" title="struct tartan_acpi::aml::AMLTable">AMLTable</a>&lt;'_&gt;, <a class="struct" href="../../../tartan_parsers/error/struct.ErrorWithPosition.html" title="struct tartan_parsers::error::ErrorWithPosition">ErrorWithPosition</a>&lt;'_, <a class="struct" href="../../../tartan_acpi/aml/parse/state/struct.ParserState.html" title="struct tartan_acpi::aml::parse::state::ParserState">ParserState</a>&lt;'_&gt;&gt;&gt;</pre><div class="docblock"><p>Parse an ACPI table that contains AML, including the header</p>
<h1 id="errors" class="section-header"><a href="#errors">Errors</a></h1>
<p>Returns an error if the AML contents cannot be parsed.</p>
</div></section><section id="search" class="content hidden"></section><section class="footer"></section><div id="rustdoc-vars" data-root-path="../../../" data-current-crate="tartan_acpi"></div>
    <script src="../../../main.js"></script><script defer src="../../../search-index.js"></script></body></html>