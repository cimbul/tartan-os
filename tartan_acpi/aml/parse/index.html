<!DOCTYPE html><html lang="en"><head><meta charset="utf-8"><meta name="viewport" content="width=device-width, initial-scale=1.0"><meta name="generator" content="rustdoc"><meta name="description" content="Parsers for AML bytecode."><meta name="keywords" content="rust, rustlang, rust-lang, parse"><title>tartan_acpi::aml::parse - Rust</title><link rel="preload" as="font" type="font/woff2" crossorigin href="../../../SourceSerif4-Regular.ttf.woff2"><link rel="preload" as="font" type="font/woff2" crossorigin href="../../../FiraSans-Regular.woff2"><link rel="preload" as="font" type="font/woff2" crossorigin href="../../../FiraSans-Medium.woff2"><link rel="preload" as="font" type="font/woff2" crossorigin href="../../../SourceCodePro-Regular.ttf.woff2"><link rel="preload" as="font" type="font/woff2" crossorigin href="../../../SourceSerif4-Bold.ttf.woff2"><link rel="preload" as="font" type="font/woff2" crossorigin href="../../../SourceCodePro-Semibold.ttf.woff2"><link rel="stylesheet" href="../../../normalize.css"><link rel="stylesheet" href="../../../rustdoc.css" id="mainThemeStyle"><link rel="stylesheet" href="../../../ayu.css" disabled><link rel="stylesheet" href="../../../dark.css" disabled><link rel="stylesheet" href="../../../light.css" id="themeStyle"><script id="default-settings" ></script><script src="../../../storage.js"></script><script defer src="../../../main.js"></script><noscript><link rel="stylesheet" href="../../../noscript.css"></noscript><link rel="alternate icon" type="image/png" href="../../../favicon-16x16.png"><link rel="alternate icon" type="image/png" href="../../../favicon-32x32.png"><link rel="icon" type="image/svg+xml" href="../../../favicon.svg"></head><body class="rustdoc mod"><!--[if lte IE 11]><div class="warning">This old browser is unsupported and will most likely display funky things.</div><![endif]--><nav class="mobile-topbar"><button class="sidebar-menu-toggle">&#9776;</button><a class="sidebar-logo" href="../../../tartan_acpi/index.html"><div class="logo-container"><img class="rust-logo" src="../../../rust-logo.svg" alt="logo"></div></a><h2 class="location"></h2></nav><nav class="sidebar"><a class="sidebar-logo" href="../../../tartan_acpi/index.html"><div class="logo-container"><img class="rust-logo" src="../../../rust-logo.svg" alt="logo"></div></a><h2 class="location"><a href="#">Module parse</a></h2><div class="sidebar-elems"><section><div class="block"><ul><li><a href="#modules">Modules</a></li><li><a href="#traits">Traits</a></li><li><a href="#functions">Functions</a></li><li><a href="#types">Type Definitions</a></li></ul></div></section></div></nav><main><div class="width-limiter"><div class="sub-container"><a class="sub-logo-container" href="../../../tartan_acpi/index.html"><img class="rust-logo" src="../../../rust-logo.svg" alt="logo"></a><nav class="sub"><form class="search-form"><div class="search-container"><span></span><input class="search-input" name="search" autocomplete="off" spellcheck="false" placeholder="Click or press ‘S’ to search, ‘?’ for more options…" type="search"><div id="help-button" title="help" tabindex="-1"><button type="button">?</button></div><div id="settings-menu" tabindex="-1"><a href="../../../settings.html" title="settings"><img width="22" height="22" alt="Change settings" src="../../../wheel.svg"></a></div></div></form></nav></div><section id="main-content" class="content"><div class="main-heading"><h1 class="fqn"><span class="in-band">Module <a href="../../index.html">tartan_acpi</a>::<wbr><a href="../index.html">aml</a>::<wbr><a class="mod" href="#">parse</a><button id="copy-path" onclick="copy_path(this)" title="Copy item path to clipboard"><img src="../../../clipboard.svg" width="19" height="18" alt="Copy item path"></button></span></h1><span class="out-of-band"><a class="srclink" href="../../../src/tartan_acpi/aml/parse.rs.html#1-2954">source</a> · <a id="toggle-all-docs" href="javascript:void(0)" title="collapse all docs">[<span class="inner">&#x2212;</span>]</a></span></div><details class="rustdoc-toggle top-doc" open><summary class="hideme"><span>Expand description</span></summary><div class="docblock"><p>Parsers for AML bytecode.</p>
<p>AML bytecode grammar is defined in §20.2. The top-level production is:</p>
<div class="example-wrap"><pre class="language-text"><code>AMLCode := DefBlockHeader TermList</code></pre></div>
<p>Everything in <code>DefBlockHeader</code> just repeats the fixed
<a href="../../struct.DescriptionHeader.html"><code>DescriptionHeader</code></a> struct used in other tables,
so there’s no reason to write a parser for it.</p>
</div></details><h2 id="modules" class="small-section-header"><a href="#modules">Modules</a></h2><div class="item-table"><div class="item-row"><div class="item-left module-item"><a class="mod" href="data/index.html" title="tartan_acpi::aml::parse::data mod">data</a></div><div class="item-right docblock-short">Data objects, defined in §20.2.3</div></div><div class="item-row"><div class="item-left module-item"><a class="mod" href="misc/index.html" title="tartan_acpi::aml::parse::misc mod">misc</a></div><div class="item-right docblock-short">Miscellaneous objects, defined in §20.2.6</div></div><div class="item-row"><div class="item-left module-item"><a class="mod" href="name/index.html" title="tartan_acpi::aml::parse::name mod">name</a></div><div class="item-right docblock-short">Name objects, defined in §20.2.2</div></div><div class="item-row"><div class="item-left module-item"><a class="mod" href="state/index.html" title="tartan_acpi::aml::parse::state mod">state</a></div><div class="item-right docblock-short">Parser context needed to disambiguate grammar</div></div><div class="item-row"><div class="item-left module-item"><a class="mod" href="term/index.html" title="tartan_acpi::aml::parse::term mod">term</a></div><div class="item-right docblock-short">Terms, defined in §20.2.5</div></div></div><h2 id="traits" class="small-section-header"><a href="#traits">Traits</a></h2><div class="item-table"><div class="item-row"><div class="item-left module-item"><a class="trait" href="trait.AMLParseError.html" title="tartan_acpi::aml::parse::AMLParseError trait">AMLParseError</a></div><div class="item-right docblock-short">Trait “alias” for error type used by AML parsers</div></div><div class="item-row"><div class="item-left module-item"><a class="trait" href="trait.Parse.html" title="tartan_acpi::aml::parse::Parse trait">Parse</a></div><div class="item-right docblock-short">An object that can be parsed from AML bytecode</div></div></div><h2 id="functions" class="small-section-header"><a href="#functions">Functions</a></h2><div class="item-table"><div class="item-row"><div class="item-left module-item"><a class="fn" href="fn.parse_table.html" title="tartan_acpi::aml::parse::parse_table fn">parse_table</a></div><div class="item-right docblock-short">Parse an ACPI table that contains AML, including the header</div></div></div><h2 id="types" class="small-section-header"><a href="#types">Type Definitions</a></h2><div class="item-table"><div class="item-row"><div class="item-left module-item"><a class="type" href="type.AMLParseResult.html" title="tartan_acpi::aml::parse::AMLParseResult type">AMLParseResult</a></div><div class="item-right docblock-short">Alias for type returned by AML parsers</div></div></div></section></div></main><div id="rustdoc-vars" data-root-path="../../../" data-current-crate="tartan_acpi" data-themes="ayu,dark,light" data-resource-suffix="" data-rustdoc-version="1.66.0-nightly (c97d02cdb 2022-10-05)" ></div></body></html>