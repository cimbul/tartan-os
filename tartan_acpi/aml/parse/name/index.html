<!DOCTYPE html><html lang="en"><head><meta charset="utf-8"><meta name="viewport" content="width=device-width, initial-scale=1.0"><meta name="generator" content="rustdoc"><meta name="description" content="Name objects, defined in §20.2.2"><title>tartan_acpi::aml::parse::name - Rust</title><script>if(window.location.protocol!=="file:")document.head.insertAdjacentHTML("beforeend","SourceSerif4-Regular-46f98efaafac5295.ttf.woff2,FiraSans-Regular-018c141bf0843ffd.woff2,FiraSans-Medium-8f9a781e4970d388.woff2,SourceCodePro-Regular-562dcc5011b6de7d.ttf.woff2,SourceCodePro-Semibold-d899c5a5c4aeb14a.ttf.woff2".split(",").map(f=>`<link rel="preload" as="font" type="font/woff2" crossorigin href="../../../../static.files/${f}">`).join(""))</script><link rel="stylesheet" href="../../../../static.files/normalize-76eba96aa4d2e634.css"><link rel="stylesheet" href="../../../../static.files/rustdoc-b25d77b18a16a9a4.css"><meta name="rustdoc-vars" data-root-path="../../../../" data-static-root-path="../../../../static.files/" data-current-crate="tartan_acpi" data-themes="" data-resource-suffix="" data-rustdoc-version="1.82.0-nightly (506052d49 2024-08-16)" data-channel="nightly" data-search-js="search-ef54e092fea9c776.js" data-settings-js="settings-4313503d2e1961c2.js" ><script src="../../../../static.files/storage-118b08c4c78b968e.js"></script><script defer src="../sidebar-items.js"></script><script defer src="../../../../static.files/main-cb0df477c2d67d00.js"></script><noscript><link rel="stylesheet" href="../../../../static.files/noscript-3b12f09e550e0385.css"></noscript><link rel="alternate icon" type="image/png" href="../../../../static.files/favicon-32x32-422f7d1d52889060.png"><link rel="icon" type="image/svg+xml" href="../../../../static.files/favicon-2c020d218678b618.svg"></head><body class="rustdoc mod"><!--[if lte IE 11]><div class="warning">This old browser is unsupported and will most likely display funky things.</div><![endif]--><nav class="mobile-topbar"><button class="sidebar-menu-toggle" title="show sidebar"></button></nav><nav class="sidebar"><div class="sidebar-crate"><h2><a href="../../../../tartan_acpi/index.html">tartan_<wbr>acpi</a><span class="version">0.1.0</span></h2></div><h2 class="location"><a href="#">Module name</a></h2><div class="sidebar-elems"><section><ul class="block"><li><a href="#functions">Functions</a></li></ul></section><h2><a href="../index.html">In tartan_<wbr>acpi::<wbr>aml::<wbr>parse</a></h2></div></nav><div class="sidebar-resizer"></div><main><div class="width-limiter"><rustdoc-search></rustdoc-search><section id="main-content" class="content"><div class="main-heading"><h1>Module <a href="../../../index.html">tartan_acpi</a>::<wbr><a href="../../index.html">aml</a>::<wbr><a href="../index.html">parse</a>::<wbr><a class="mod" href="#">name</a><button id="copy-path" title="Copy item path to clipboard">Copy item path</button></h1><span class="out-of-band"><a class="src" href="../../../../src/tartan_acpi/aml/parse.rs.html#422">source</a> · <button id="toggle-all-docs" title="collapse all docs">[<span>&#x2212;</span>]</button></span></div><details class="toggle top-doc" open><summary class="hideme"><span>Expand description</span></summary><div class="docblock"><p>Name objects, defined in §20.2.2</p>
</div></details><h2 id="functions" class="section-header">Functions<a href="#functions" class="anchor">§</a></h2><ul class="item-table"><li><div class="item-name"><a class="fn" href="fn.parse_target.html" title="fn tartan_acpi::aml::parse::name::parse_target">parse_<wbr>target</a></div><div class="desc docblock-short">Grammar:</div></li><li><div class="item-name"><a class="fn" href="fn.super_name_minus_invoke.html" title="fn tartan_acpi::aml::parse::name::super_name_minus_invoke">super_<wbr>name_<wbr>minus_<wbr>invoke</a></div><div class="desc docblock-short">Parse a <a href="../../name/enum.SuperName.html" title="enum tartan_acpi::aml::name::SuperName"><code>SuperName</code></a>, but do <em>not</em> allow method invocations.</div></li></ul></section></div></main></body></html>