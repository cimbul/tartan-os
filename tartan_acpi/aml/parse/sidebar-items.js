initSidebarItems({"fn":[["parse_table","Parse an ACPI table that contains AML, including the header"]],"mod":[["data","Data objects, defined in §20.2.3"],["misc","Miscellaneous objects, defined in §20.2.6"],["name","Name objects, defined in §20.2.2"],["state","Parser context needed to disambiguate grammar"],["term","Terms, defined in §20.2.5"]],"trait":[["AMLParseError","Trait “alias” for error type used by AML parsers"],["Parse","An object that can be parsed from AML bytecode"]],"type":[["AMLParseResult","Alias for type returned by AML parsers"]]});