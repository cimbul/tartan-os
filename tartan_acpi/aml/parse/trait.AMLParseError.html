<!DOCTYPE html><html lang="en"><head><meta charset="utf-8"><meta name="viewport" content="width=device-width, initial-scale=1.0"><meta name="generator" content="rustdoc"><meta name="description" content="Trait “alias” for error type used by AML parsers"><title>AMLParseError in tartan_acpi::aml::parse - Rust</title><script>if(window.location.protocol!=="file:")document.head.insertAdjacentHTML("beforeend","SourceSerif4-Regular-6b053e98.ttf.woff2,FiraSans-Italic-81dc35de.woff2,FiraSans-Regular-0fe48ade.woff2,FiraSans-MediumItalic-ccf7e434.woff2,FiraSans-Medium-e1aa3f0a.woff2,SourceCodePro-Regular-8badfe75.ttf.woff2,SourceCodePro-Semibold-aa29a496.ttf.woff2".split(",").map(f=>`<link rel="preload" as="font" type="font/woff2" crossorigin href="../../../static.files/${f}">`).join(""))</script><link rel="stylesheet" href="../../../static.files/normalize-9960930a.css"><link rel="stylesheet" href="../../../static.files/rustdoc-cf3c48c1.css"><meta name="rustdoc-vars" data-root-path="../../../" data-static-root-path="../../../static.files/" data-current-crate="tartan_acpi" data-themes="" data-resource-suffix="" data-rustdoc-version="1.86.0-nightly (854f22563 2025-01-31)" data-channel="nightly" data-search-js="search-2d513d54.js" data-settings-js="settings-6dad6058.js" ><script src="../../../static.files/storage-302de22f.js"></script><script defer src="sidebar-items.js"></script><script defer src="../../../static.files/main-9b5d7e41.js"></script><noscript><link rel="stylesheet" href="../../../static.files/noscript-893ab5e7.css"></noscript><link rel="alternate icon" type="image/png" href="../../../static.files/favicon-32x32-6580c154.png"><link rel="icon" type="image/svg+xml" href="../../../static.files/favicon-044be391.svg"></head><body class="rustdoc trait"><!--[if lte IE 11]><div class="warning">This old browser is unsupported and will most likely display funky things.</div><![endif]--><nav class="mobile-topbar"><button class="sidebar-menu-toggle" title="show sidebar"></button></nav><nav class="sidebar"><div class="sidebar-crate"><h2><a href="../../../tartan_acpi/index.html">tartan_<wbr>acpi</a><span class="version">0.1.0</span></h2></div><div class="sidebar-elems"><section id="rustdoc-toc"><h2 class="location"><a href="#">AMLParse<wbr>Error</a></h2><h3><a href="#dyn-compatibility">Dyn Compatibility</a></h3><h3><a href="#implementors">Implementors</a></h3></section><div id="rustdoc-modnav"><h2><a href="index.html">In tartan_<wbr>acpi::<wbr>aml::<wbr>parse</a></h2></div></div></nav><div class="sidebar-resizer"></div><main><div class="width-limiter"><rustdoc-search></rustdoc-search><section id="main-content" class="content"><div class="main-heading"><div class="rustdoc-breadcrumbs"><a href="../../index.html">tartan_acpi</a>::<wbr><a href="../index.html">aml</a>::<wbr><a href="index.html">parse</a></div><h1>Trait <span class="trait">AMLParseError</span><button id="copy-path" title="Copy item path to clipboard">Copy item path</button></h1><rustdoc-toolbar></rustdoc-toolbar><span class="sub-heading"><a class="src" href="../../../src/tartan_acpi/aml/parse.rs.html#95">Source</a> </span></div><pre class="rust item-decl"><code>pub trait AMLParseError&lt;'a&gt;: <a class="trait" href="state/trait.ReplaceableParseError.html" title="trait tartan_acpi::aml::parse::state::ReplaceableParseError">ReplaceableParseError</a>&lt;&amp;'a [<a class="primitive" href="https://doc.rust-lang.org/nightly/core/primitive.u8.html">u8</a>], <a class="struct" href="state/struct.ParserState.html" title="struct tartan_acpi::aml::parse::state::ParserState">ParserState</a>&lt;'a&gt;&gt; { }</code></pre><details class="toggle top-doc" open><summary class="hideme"><span>Expand description</span></summary><div class="docblock"><p>Trait “alias” for error type used by AML parsers</p>
</div></details><h2 id="dyn-compatibility" class="section-header">Dyn Compatibility<a href="#dyn-compatibility" class="anchor">§</a></h2><div class="dyn-compatibility-info"><p>This trait is <b>not</b> <a href="https://doc.rust-lang.org/nightly/reference/items/traits.html#dyn-compatibility">dyn compatible</a>.</p><p><i>In older versions of Rust, dyn compatibility was called "object safety", so this trait is not object safe.</i></p></div><h2 id="implementors" class="section-header">Implementors<a href="#implementors" class="anchor">§</a></h2><div id="implementors-list"><section id="impl-AMLParseError%3C'a%3E-for-T" class="impl"><a class="src rightside" href="../../../src/tartan_acpi/aml/parse.rs.html#97-100">Source</a><a href="#impl-AMLParseError%3C'a%3E-for-T" class="anchor">§</a><h3 class="code-header">impl&lt;'a, T&gt; <a class="trait" href="trait.AMLParseError.html" title="trait tartan_acpi::aml::parse::AMLParseError">AMLParseError</a>&lt;'a&gt; for T<div class="where">where
    T: <a class="trait" href="state/trait.ReplaceableParseError.html" title="trait tartan_acpi::aml::parse::state::ReplaceableParseError">ReplaceableParseError</a>&lt;&amp;'a [<a class="primitive" href="https://doc.rust-lang.org/nightly/core/primitive.u8.html">u8</a>], <a class="struct" href="state/struct.ParserState.html" title="struct tartan_acpi::aml::parse::state::ParserState">ParserState</a>&lt;'a&gt;&gt;,</div></h3></section></div><script src="../../../trait.impl/tartan_acpi/aml/parse/trait.AMLParseError.js" async></script></section></div></main></body></html>