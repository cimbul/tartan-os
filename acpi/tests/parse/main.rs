#![warn(clippy::pedantic)]

use insta::{assert_debug_snapshot, glob, with_settings};
use std::fs;
use tartan_acpi::aml::parse::parse_table;

#[test]
fn parse_all() {
    // Parse all AML files and compare to past output
    glob!("aml/*.aml", |aml_path| {
        // Make the autogenerated snapshot filenames less wordy
        let test_name = aml_path.file_stem().unwrap().to_str().unwrap();
        with_settings!({snapshot_suffix => ""}, {
            let aml_data = fs::read(aml_path).unwrap();
            assert_debug_snapshot!(test_name, parse_table(&aml_data));
        });
    });
}
