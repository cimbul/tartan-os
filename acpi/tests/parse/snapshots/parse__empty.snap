---
source: acpi/tests/parse/main.rs
expression: parse_table(&aml_data)
---
Ok(
    AMLTable {
        header: DescriptionHeader {
            signature: [
                68,
                83,
                68,
                84,
            ],
            length: 36,
            revision: 2,
            checksum: 157,
            oem_id: [
                84,
                97,
                114,
                116,
                97,
                110,
            ],
            oem_table_id: [
                69,
                109,
                112,
                116,
                121,
                0,
                0,
                0,
            ],
            oem_revision: 0,
            creator_id: 1280593481,
            creator_revision: 538969879,
        },
        body: [],
    },
)
